src/components/AssetSearch/AssetList.tsx

```diff
@@ -23,17 +24,69 @@ type ItemData<T> = {
   handleClick: T
 }
 
+const sortAssetsByAccountAndMarketCap = (
+  assets: Asset[],
+  enrichedData: Record<string, { fiatAmount: string; cryptoAmount: string; marketCap: string }>,
+): Asset[] => {
+  return sortBy(assets, [
+    asset => {
+      return Number(enrichedData[asset.assetId].fiatAmount)
+    },
+    asset => {
+      return Number(enrichedData[asset.assetId].marketCap)
+    },
+  ]).reverse()
+}
+
+/**
+ * This function create a data object(cryptoAmount, fiatAmount, marketCap) from the Asset, portfolioAcountRows and marketData for each asset for facilitate sorting.
+ * @param assets
+ * @param portfolioAcountRows
+ * @param marketData
+ * @returns Record<string, { fiatAmount: string; cryptoAmount: string; marketCap: string }>
+ */
+export const createEnrichData = (
+  assets: Asset[],
+  portfolioAcountRows: AccountRowData[],
+  marketData: {
+    [x: string]: MarketData | undefined
+  },
+): Record<string, { fiatAmount: string; cryptoAmount: string; marketCap: string }> => {
+  const result = {} as Record<
+    string,
+    { fiatAmount: string; cryptoAmount: string; marketCap: string }
+  >
+  assets.forEach(asset => {
```

Comment:
If you chain a `.map` or `.reduce` instead of a `.forEach`, you won't need to have an initial empty object, and that will solve the typings issues that made you cast `result` in the first place

Reply:
I removed the cast at the beginning of result and simply created the constant with the type I defined.